tests/generation/test_utils.py

```diff
@@ -440,28 +401,26 @@ def _contrastive_generate(
     @pytest.mark.generate
     def test_greedy_generate(self):
         for model_class in self.all_generative_model_classes:
-            config, input_ids, attention_mask, inputs_dict = self._get_input_ids_and_config()
+            config, inputs_dict = self.prepare_config_and_inputs_for_generate()
+            main_input = inputs_dict[self.input_name]
```

Comment:
wondering what happens if we use `model_cls.main_input_name`? AFAIR from few months ago, there were some inconsistencies in how model main input is defined, and we could make another round of cleaning up on that because `main_input_name` is also used in `generate()`. Maybe we can have a more generalized interface and testing suite?

Reply:
Not only it worked, but also it allowed us to remove the `input_name` attribute from all testers 💛